akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/EventConsumerServiceImpl.scala

```diff
@@ -0,0 +1,162 @@
+                  })
+                case None =>
+                  logger.info2("Event stream from [{}] for stream id [{}] started", init.originId, init.streamId)
+                  eventsAndFiltered
+              }
+
+            case None =>
+              logger.debug2(
+                "Event producer [{}] wanted to push events for stream id [{}] but that is not among the accepted stream ids",
+                init.originId,
+                init.streamId)
+              throw new GrpcServiceException(Status.PERMISSION_DENIED.withDescription(
+                s"Events for stream id [${init.streamId}] not accepted by this consumer"))
+          }
+
+        case (_, _) =>
+          throw new GrpcServiceException(Status.INVALID_ARGUMENT.withDescription(
+            "Consumer stream in must start with Init event followed by events but got something else"))
+      }
+      .map { consumeEventIn =>
+        if (consumeEventIn.message.isEvent)
+          ProtobufProtocolConversions.eventToEnvelope[Any](consumeEventIn.getEvent, protoAnySerialization)
+        else if (consumeEventIn.message.isFilteredEvent) {
+          ProtobufProtocolConversions.filteredEventToEnvelope[Any](consumeEventIn.getFilteredEvent)
+        } else {
+          throw new GrpcServiceException(Status.INVALID_ARGUMENT
+            .withDescription(s"Unexpected type of ConsumeEventIn: ${consumeEventIn.message.getClass}"))
+        }
+      }
+      // FIXME config for parallelism, and perPartition (aligned with event writer batch config)
+      // Note that perPartition must be something like half the siez of the event writer max buffer as the previous
+      // in flight batch will complete one by one
+      .mapAsyncPartitioned(parallelism, perPartitionParallelism)(_.persistenceId) { (originalEnvelope, _) =>
+        val (destination, transformer) = destinationAndTransformerForStream
+        val transformedEventEnvelope = transformer(originalEnvelope)
+        if (logger.isTraceEnabled)
+          logger.traceN(
+            "Saw event [{}] for pid [{}]{}",
+            transformedEventEnvelope.sequenceNr,
+            transformedEventEnvelope.persistenceId,
+            if (transformedEventEnvelope.filtered) " filtered" else "")
+
+        destination.eventWriter
+          .askWithStatus[EventWriter.WriteAck](EventWriter.Write(
+            transformedEventEnvelope.persistenceId,
+            transformedEventEnvelope.sequenceNr,
+            transformedEventEnvelope.eventOption.getOrElse(FilteredPayload),
+            transformedEventEnvelope.eventMetadata,
+            transformedEventEnvelope.tags,
+            _))
+          .map(_ =>
+            // ack using the original pid in case it was transformed
+            ConsumeEventOut(ConsumeEventOut.Message.Ack(
+              ConsumerEventAck(originalEnvelope.persistenceId, originalEnvelope.sequenceNr))))(
+            ExecutionContexts.parasitic)
+          .recover {
+            case NonFatal(ex) =>
```

Comment:
`NonFatal` is not needed in Future recover, it will only expose exceptions that are NonFatal